src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -463,48 +739,244 @@ function saturation_adjustment(e_int, ρ, q_tot, T_init = T_triple)
   end
 end
 
+
 """
-    liquid_ice_pottemp(T, p[, q_tot=0, q_liq=0, q_ice=0])
+    saturation_adjustment_q_t_θ_l(p, q_tot, θ_liq)
+
+Compute the temperature that is consistent with
+
+ - `p` pressure
+ - `q_tot` total specific humidity
+ - `θ_liq` liquid potential temperature
+
+See also [`saturation_adjustment`](@ref).
+"""
+function saturation_adjustment_q_t_θ_l(θ_liq::DT, q_tot::DT, ρ::DT, p::DT) where DT
```

Comment:
I think a preferable way of writing this function is to make it parallel to the saturation adjustment for internal energy, just replacing the roots equation and the initial temperature bracket.